kernel/src/cpu/idt/common.rs

```diff
@@ -186,20 +196,14 @@ impl IDT {
         unsafe {
             asm!("lidt (%rax)", in("rax") &desc, options(att_syntax));
         }
-
-        self
     }
+}
 
+impl ReadLockGuard<'static, IDT> {
     pub fn base_limit(&self) -> (u64, u32) {
-        let base = (self as *const IDT) as u64;
+        let base: *const IDT = core::ptr::from_ref(self);
```

Comment:
```suggestion
        let base: *const IDT = self;
```

Unless your intention is to be explicit, `core::ptr::from_ref` isn't required for converting a reference to its pointer type.

Reply:
Yes, the idea was to be explicit in the conversion from `&ReadLockGuard<'static, IDT>` to `&IDT` to `*const IDT`.